src/tosam.py

```diff
@@ -1,5 +1,9 @@
 import argparse
 
+# HELP PLEASE! I don't understand what format this is supposed to be outputted in.
+# From the test-tosam.sh file i gather that the result will be compared to 
+# mississippi.sam. The sam-file has an extra column with the CIGAR which isn't
+# available in mississippi.mas. I'm not sure what to do.
```

Comment:
If you write a comment on the **Pull requests** > **Feedback** page with `@mailund` in it, I will get the notification.

<img width="780" alt="Screenshot 2022-09-01 at 11 26 59" src="https://user-images.githubusercontent.com/3206538/187880901-61cde164-eb21-4e38-a9c6-f7d04933d20d.png">

Writing a comment or a commit message will not send me the same notification, so I will get around to looking at it much slower, or not at all, when we are later in the course. For the first couple of weeks, I will try to read all the code but will prioritise notifications where I am mentioned. Later, I will not have the time to read everything but will do it at random, but always get to those where I am mentioned.

Anyway, I got here in the end.

The CIGAR for this exercise was explained in the problem description:

> The CIGAR is always the length of the pattern followed by M where there are no edits, and at this point in the class we do not worry about approximative matching.

The CIGAR explains how you should modify one string to become another, but that only applies for approximative (almost) matches. We need that later in the class. For now, we have to make a CIGAR that says we don't change anything. That means each character matches (symbol `M`) character by character, so if you have the string `iss` and have to match all three characters, then it is `MMM`. This is run-length encoded to become `3M`.

For your code, you can get the cigar as

```python
   cigar = f"{len(read_str)}M"
```